File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque
input = sys.stdin.readline
n = int(input())
qu = deque()
for i in range(n):
    text = input().split()
    if text[0]=='push':
        qu.append(text[1])
    elif text[0] =='pop':
        if len(qu)==0:
            print(-1)
        else:
            print(qu.popleft())
    elif text[0] =='size':
        print(len(qu))
    elif text[0]=='empty':
        if len(qu)==0:
            print(1)
        else:
            print(0)
    elif text[0] =='front':
        if len(qu)==0:
            print(-1)
        else:
            print(qu[0])
    elif text[0]=='back':
        if len(qu)==0:
            print(-1)
        else:
            print(qu[-1])

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Queue<Integer> queue = new LinkedList<>();
        int last = 0; // 마지막 요소를 추적하기 위한 변수

        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            String[] command = br.readLine().split(" ");
            
            switch (command[0]) {
                case "push":
                    last = Integer.parseInt(command[1]);
                    queue.offer(last);
                    break;
                case "pop":
                    bw.write(queue.isEmpty() ? "-1" : queue.poll().toString());
                    bw.newLine();
                    break;
                case "size":
                    bw.write(String.valueOf(queue.size()));
                    bw.newLine();
                    break;
                case "empty":
                    bw.write(queue.isEmpty() ? "1" : "0");
                    bw.newLine();
                    break;
                case "front":
                    bw.write(queue.isEmpty() ? "-1" : queue.peek().toString());
                    bw.newLine();
                    break;
                case "back":
                    bw.write(queue.isEmpty() ? "-1" : String.valueOf(last));
                    bw.newLine();
                    break;
            }
        }
        
        bw.flush();
        bw.close();
        br.close();
    }
}